### PR TITLE
Adding anonymous users and fixing representations

### PR DESCRIPTION
**Pull request by VanDavv**
PreAuthFilter now works correctly and representations have correct order

### CODE REVIEW

File: backend/app/studybox-h2.yml [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
+    port: 2000
+
 database:
   driverClass: org.h2.Driver
   user: sa
   password: sa
-  url: jdbc:h2:./target/studybox.db
+  url: jdbc:h2:./studybox
 
 authHeaderRequired: false
```
Comment (mlotysz) [inline backend/app/studybox-h2.yml]: remove

Comment (VanDavv) [inline backend/app/studybox-h2.yml]: This parameter?

Comment (mlotysz) [inline backend/app/studybox-h2.yml]: Am I right, that this parameter is no longer needed: `authHeaderRequired`? We know that it's always `false` and we can remove it to not pollute source code with no needed code (KISS).

File: backend/app/studybox-h2.yml [hunk 2]
```diff
@@ -1,7 +1,16 @@
+server:
```
Comment (mlotysz) [inline backend/app/studybox-h2.yml]: Copy this server block to other configuration/s.

**Comment by mlotysz** [comment]
Thanks!

**Pull request merged**